### ISSUE TITLE
jinja rule seems to not work

### ISSUE THREAD

**Issue by Malyuk-A** [ansible/ansible-lint#4539]
##### Summary

The [jinja rule](https://ansible.readthedocs.io/projects/lint/rules/jinja/) does not seem to work at all - i.e. it does not detect any errors/warnings.

Not even on the officially recommended ansible-dev-tools containers using the official erroneous example of code -> See examples below.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint 25.1.2 using ansible-core:2.18.2 ansible-compat:25.1.2 ruamel-yaml:0.18.10 ruamel-yaml-clib:0.2.12
```
After updating pip and ansible-lint same occurs on

```console
ansible-lint 25.1.3 using ansible-core:2.18.2 ansible-compat:25.1.4 ruamel-yaml:0.18.10 ruamel-yaml-clib:0.2.12
```

System/Container Specs:
```console
          /:-------------:\          root@59b154eae922 
       :-------------------::        ----------------- 
     :-----------/shhOHbmp---:\      OS: Fedora Linux 41 (Container Image) x86_64 
   /-----------omMMMNNNMMD  ---:     Host: VirtualBox 1.2 
  :-----------sMMMMNMNMP.    ---:    Kernel: 6.1.0-22-amd64 
 :-----------:MMMdP-------    ---\   Uptime: 9 hours, 26 mins 
,------------:MMMd--------    ---:   Packages: 337 (rpm) 
:------------:MMMd-------    .---:   Shell: zsh 5.9 
:----    oNMMMMMMMMMNho     .----:   Resolution: 1920x969 
:--     .+shhhMMMmhhy++   .------/   Terminal: dumb-init 
:-    -------:MMMd--------------:    CPU: 12th Gen Intel i7-12700F (1) @ 2.112GHz 
:-   --------/MMMd-------------;     Memory: 2668MiB / 3915MiB 
:-    ------/hMMMy------------:
:-- :dMNdhhdNMMNo------------;                               
:---:sdNMMMMNds:------------:                                
:------:://:-------------::
:---------------------://
```

##### STEPS TO REPRODUCE

```console
docker pull ghcr.io/ansible/community-ansible-dev-tools
```
```console
docker run -it ghcr.io/ansible/community-ansible-dev-tools bash
```
How it should work: https://ansible.readthedocs.io/projects/lint/rules/fqcn/#problematic-code

```console
bash-5.2# bat fqcn_wrong_code.yml
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: fqcn_wrong_code.yml
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ ---
   2   │ - name: Example playbook
   3   │   hosts: all
   4   │   tasks:
   5   │     - name: Create an SSH connection
   6   │       shell: ssh ssh_user@{{ ansible_ssh_host }} # <- Does not use the FQCN for the shell module.
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
bash-5.2# ansible-lint --strict fqcn_wrong_code.yml
WARNING  Listing 3 violation(s) that are fatal
command-instead-of-shell: Use shell only when shell functionality is required.
fqcn_wrong_code.yml:5 Task/Handler: Create an SSH connection

fqcn[action-core]: Use FQCN for builtin module actions (shell).
fqcn_wrong_code.yml:5 Use `ansible.builtin.shell` or `ansible.legacy.shell` instead.

no-changed-when: Commands should not change things if nothing needs doing.
fqcn_wrong_code.yml:5 Task/Handler: Create an SSH connection

Read documentation for instructions on how to ignore specific rule violations.

# Rule Violation Summary

  1 command-instead-of-shell profile:basic tags:command-shell,idiom
  1 no-changed-when profile:basic tags:command-shell,idempotency
  1 fqcn profile:basic tags:formatting

Failed: 3 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'min'.
```

How this is not working for the `jinja` rule: https://ansible.readthedocs.io/projects/lint/rules/jinja/#problematic-code
```console
bash-5.2# bat jinja_wrong_code.yml
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: jinja_wrong_code.yml
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ ---
   2   │ - name: Some task
   3   │   vars:
   4   │     foo: "{{some|dict2items}}" # <-- jinja[spacing]
   5   │     bar: "{{ & }}" # <-- jinja[invalid]
   6   │   when: "{{ foo | bool }}" # <-- jinja[spacing] - 'when' has implicit templating
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
bash-5.2# ansible-lint --strict jinja_wrong_code.yml

Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
bash-5.2# 
```

Create the two files and run ansible-lint on them to verify.

##### Desired Behavior

That the "erroneous file" with the official `jinja` rule  should also properly detect the errors or warnings.

##### Actual Behavior
No errors/warning are detected.
